tests/test-update-broken-delta.c

```diff
@@ -0,0 +1,127 @@
+ *
+ * Copyright © 2017 Kinvolk GmbH
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * Authors:
+ *  - Krzesimir Nowak <krzesimir@kinvolk.io>
+ */
+
+#include "misc-utils.h"
+#include "spawn-utils.h"
+#include "ostree-spawn.h"
+#include "eos-test-utils.h"
+#include "eos-test-convenience.h"
+
+#include <gio/gio.h>
+#include <locale.h>
+#include <string.h>
+
+static void
+delete_big_file_object_from_client_repo (EtcData *data)
+{
+  const gchar *bigfile_path = eos_test_client_get_big_file_path ();
+  const gchar *paths[] =
+    {
+      bigfile_path,
+      NULL
+    };
+  g_autoptr(GFile) client_repo = NULL;
+  OstreeLsFlags flags = OSTREE_LS_CHECKSUM;
+  g_auto(CmdResult) listed = CMD_RESULT_CLEARED;
+  g_autoptr(GError) error = NULL;
+  g_autofree gchar *escaped_bigfile_path = NULL;
+  g_autofree gchar *regex = NULL;
+  g_autoptr(GRegex) gregex = NULL;
+  gboolean matched;
+  g_autoptr(GMatchInfo) match_info = NULL;
+  g_autofree gchar *checksum = NULL;
+  g_autofree gchar *object = NULL;
+
+  g_assert_nonnull (data);
+  g_assert_nonnull (data->client);
+
+  client_repo = eos_test_client_get_repo (data->client);
+  ostree_ls (client_repo,
+             flags,
+             default_ref,
+             paths,
+             &listed,
+             &error);
+  g_assert_no_error (error);
+  g_assert_true (cmd_result_ensure_ok_verbose (&listed));
+
+  escaped_bigfile_path = g_regex_escape_string (bigfile_path, -1);
+  regex = g_strdup_printf ("\\s+([0-9a-zA-Z]{64})\\s+%s", escaped_bigfile_path);
+  gregex = g_regex_new (regex,
+                        /* compile flags: */ 0,
+                        /* match flags: */ 0,
+                        &error);
+  g_assert_no_error (error);
+  matched = g_regex_match (gregex,
+                           listed.standard_output,
+                           /* match flags: */ 0,
+                           &match_info);
+  g_assert_true (matched);
+  checksum = g_match_info_fetch (match_info,
+                                 /* match group: */ 1);
+  object = g_strdup_printf ("%s.file", checksum);
+  etc_delete_object (client_repo, object);
+}
+
+static void
+test_update_broken_delta (EosUpdaterFixture *fixture,
+                          gconstpointer user_data)
+{
```

Comment:
Could do with a brief doc comment for the test function explaining what the test is trying to verify. Could also use `g_test_bug()`.